spec/service/cloud_watch_service_spec.rb

```diff
@@ -39,6 +40,22 @@
         unit: "Count",
       }).and_return(metric_response)
 
+      expect(cloudwatch_client).to receive(:get_metric_statistics).with({
+        metric_name: "submitted",
+        namespace: "forms/#{forms_env}",
+        dimensions: [
+          {
+            name: "form_id",
+            value: form_id.to_s,
+          },
+        ],
+        start_time: Time.zone.now.midnight - 7.days,
+        end_time: Time.zone.now.midnight,
+        period: 604_800,
+        statistics: %w[Sum],
+        unit: "Count",
+      })
```

Comment:
Entirely optional but potentially you could use:

```suggestion
      }).once
```
here (and for the week_starts test below) to make the tests a bit more strict. Doesn't look like it'll make a huge difference with the code as it stands, but could catch double requests if we try any fancy refactoring in future.

Reply:
I do like this, especially as its an external API call so hitting it too hard accidentally could spell trouble!